lib/contractor_list/bloc/search_bloc.dart

```diff
@@ -0,0 +1,44 @@
+//receives events and produces states
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:rate_my_contractor/contractor_list/domain/contractor_repository.dart';
+import 'package:rate_my_contractor/contractor_list/domain/models/contractor.dart';
+part 'search_state.dart';
+part 'search_event.dart';
+
+//have a try catch here and emit error state
+class SearchBloc extends Bloc<SearchEvent, SearchState> {
+  final ContractorRepository repository;
+
+  SearchBloc(this.repository) : super(SearchState.initial()) {
```

Comment:
Can we just create SearchState() here? We can just set the initial values you pass in this factory function as the default values for the default constructor